### PR TITLE
Change setupStore helper default serializer to JSONAPISerializer #4754

### PR DESCRIPTION
**Pull request by eshtadc**

### CODE REVIEW

File: tests/integration/filter-test.js [hunk 1]
```diff
@@ -544,7 +554,13 @@ test('it is possible to filter created records by dirtiness', function(assert) {
   run(() => {
     customAdapter(env, DS.Adapter.extend({
       createRecord(type, model, snapshot) {
-        return Ember.merge(Ember.merge({}, snapshot._attributes), { id: snapshot.id} )
+        return {
+          data: {
+            id: snapshot.id,
+            type: model.modelName,
+            attributes: Ember.merge(Ember.merge({}, snapshot._attributes))
```
Comment (sbatson5) [inline tests/integration/filter-test.js]: Do we still need to specify `Ember.merge` here twice if we aren't merging the `id`?

Comment (eshtadc) [inline tests/integration/filter-test.js]: Good point ;-)

**Comment by hjdivad** [comment]
@bmac This touches a lot of things, but is a nice cleanup.  Any objections to merging?

**Comment by bmac** [comment]
Sound good to me @hjdivad.

**Comment by hjdivad** [comment]
Thanks for your contribution @eshtadc!

**Comment by stefanpenner** [comment]
@eshtadc awesome!! 🎉  🚀

**Comment by workmanw** [comment]
Is this a breaking change? It seems like it could be, but perhaps I'm missing something.

**Comment by stefanpenner** [comment]
@workmanw I don't think so, as it merely changed the tests to use the more modern and recommended flow. But if you notice a regression, please share!

**Comment by workmanw** [comment]
Oh, oops. I didn't realize this change was for test setup. I thought it was for regular injection. 

giphy-downsized

-----------

Side note, this is a mammoth commit. 👏 👏

**Comment by stefanpenner** [comment]
@workmanw keep us honest!

**Pull request merged**